src/main/java/org/folio/circulation/infrastructure/storage/requests/RequestRepository.java

```diff
@@ -85,23 +85,22 @@ public static RequestRepository using(org.folio.circulation.support.Clients clie
 
   public CompletableFuture<Result<MultipleRecords<Request>>> findBy(String query) {
     return requestsStorageClient.getManyWithRawQueryStringParameters(query)
-      .thenApply(flatMapResult(this::mapResponseToRequests))
-      .thenComposeAsync(result -> itemRepository.fetchItemsFor(result, Request::withItem))
+      .thenApply(flatMapResult(this::mapResponseToRequests)).thenComposeAsync(this::fetchFields);
+  }
+
+  CompletableFuture<Result<MultipleRecords<Request>>> findBy(CqlQuery query, PageLimit pageLimit) {
+    return findByWithoutItems(query, pageLimit).thenComposeAsync(this::fetchFields);
```

Comment:
return findByWithoutItems(query, pageLimit)
      .thenComposeAsync(r -> r.after(this::fetchFields));

Reply:
fixed